src/btree/bt_read.c

```diff
@@ -536,26 +541,23 @@ __page_read(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags)
 	case WT_REF_LIMBO:
 	case WT_REF_LOOKASIDE:
 		WT_ERR(__page_read_lookaside(
-		    session, ref, previous_state, &final_state));
+		    session, ref, previous_state, &final_state, &prepared));
 		break;
 	}
 
 	/*
 	 * Once the page is instantiated, we no longer need the history in
 	 * lookaside.  We leave the lookaside sweep thread to do most cleanup,
-	 * but it can only remove keys that skew newest (if there are entries
-	 * in the lookaside newer than the page, they need to be read back into
-	 * cache or they will be lost).
-	 *
-	 * There is no reason for the lookaside remove should fail, but ignore
-	 * it if for some reason it fails, we've got a valid page.
+	 * but it can only remove committed updates and keys that skew newest
+	 * (if there are entries in the lookaside newer than the page, they need
+	 * to be read back into cache or they will be lost).
 	 *
 	 * Don't free WT_REF.page_las, there may be concurrent readers.
 	 */
 	if (final_state == WT_REF_MEM &&
-	    ref->page_las != NULL && !ref->page_las->skew_newest)
-		WT_IGNORE_RET(__wt_las_remove_block(
-		    session, ref->page_las->las_pageid, false));
+	    ref->page_las != NULL && (prepared || !ref->page_las->skew_newest))
+		WT_ERR(__wt_las_remove_block(
```

Comment:
lock_wait is always passed as true, we can remove that arguement from declaration.